yarn-project/p2p/src/client/p2p_client.test.ts

```diff
@@ -100,82 +89,43 @@ describe('In-Memory P2P Client', () => {
     expect(txPool.addTxs).toHaveBeenCalledTimes(2);
   });
 
-  it('restores the previous block number it was at', async () => {
-    await client.start();
-    const synchedBlock = await client.getSyncedLatestBlockNum();
-    await client.stop();
-
-    const client2 = new P2PClient(P2PClientType.Full, kvStore, blockSource, mempools, p2pService);
-    await expect(client2.getSyncedLatestBlockNum()).resolves.toEqual(synchedBlock);
-  });
-
   it('deletes txs once block is proven', async () => {
     blockSource.setProvenBlockNumber(0);
     await client.start();
     expect(txPool.deleteTxs).not.toHaveBeenCalled();
 
-    await advanceToProvenBlock(5);
+    advanceToProvenBlock(5);
+    await sleep(100); // wait for the event to be emitted and handled
```

Comment:
I guess this is needed because `advanceToProvenBlock` is no longer async? Slightly worried this might flake in the future, but probably fine for now :grin:

Reply:
updated